Datadog/IntegrationUnitTests/Public/Datadog+MultipleInstancesIntegrationTests.swift

```diff
@@ -0,0 +1,100 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+import TestUtilities
+@testable import DatadogCore
+import DatadogInternal
+import DatadogLogs
+
+class Datadog_MultipleInstancesIntegrationTests: XCTestCase {
+    /// The configuraiton of default instance of SDK.
+    private var defaultInstanceConfig = Datadog.Configuration(clientToken: "main-token", env: "default-env")
+    /// The configuraiton of custom instance of SDK.
+    private var customInstanceConfig = Datadog.Configuration(clientToken: "custom-token", env: "custom-env")
+
+    override func setUp() {
+        super.setUp()
+        CreateTemporaryDirectory()
+
+        // Root system directory for both instances:
+        let systemDirectory = Directory(url: temporaryDirectory)
+        defaultInstanceConfig.systemDirectory = { systemDirectory }
+        customInstanceConfig.systemDirectory = { systemDirectory }
+    }
+
+    override func tearDown() {
+        DeleteTemporaryDirectory()
+        super.tearDown()
+    }
+
+    func testGivenTwoInstancesOfSDK_whenCollectingLogs_thenEachSDKUploadsItsOwnData() throws {
+        let customInstanceName = "custom"
+        let numberOfLogs = 10
+        let defaultHTTPClient = HTTPClientMock(responseCode: 200)
+        let customHTTPClient = HTTPClientMock(responseCode: 200)
+        defaultInstanceConfig.httpClientFactory = { _ in defaultHTTPClient }
+        customInstanceConfig.httpClientFactory = { _ in customHTTPClient }
+        defaultInstanceConfig.bundle = .mockWith(bundleIdentifier: "com.bundle.default", CFBundleShortVersionString: "1.0-default")
+        customInstanceConfig.bundle = .mockWith(bundleIdentifier: "com.bundle.custom", CFBundleShortVersionString: "1.0-custom")
+
+        // Given
+        Datadog.initialize(with: defaultInstanceConfig, trackingConsent: .granted)
+        Datadog.initialize(with: customInstanceConfig, trackingConsent: .granted, instanceName: customInstanceName)
+
+        Logs.enable(with: .init())
+        Logs.enable(with: .init(), in: Datadog.sdkInstance(named: customInstanceName))
```

Comment:
is it worth checking other features? 

Specially crash reporting.

Reply:
👍 More coverage will be discussed and added after `RUMM-3514`. For now, we'd like to keep it minimal.